access/grpc/grpc.go

```diff
@@ -645,3 +657,149 @@ func (c *BaseClient) GetExecutionResultForBlockID(ctx context.Context, blockID f
+		return nil, newRPCError(err)
+	}
+
+	return messageToBlockExecutionData(ed.GetBlockExecutionData())
+
+}
+
+func (c *BaseClient) SubscribeExecutionData(
+	ctx context.Context,
+	startBlockID flow.Identifier,
+	startHeight uint64,
+	opts ...grpc.CallOption,
+) (<-chan flow.ExecutionDataStreamResponse, <-chan error, error) {
+
+	if startBlockID != flow.EmptyID && startHeight > 0 {
+		return nil, nil, fmt.Errorf("cannot specify both start block ID and start height")
+	}
+
+	req := executiondata.SubscribeExecutionDataRequest{}
+	if startBlockID != flow.EmptyID {
+		req.StartBlockId = startBlockID[:]
+	}
+	if startHeight > 0 {
+		req.StartBlockHeight = startHeight
+	}
+
+	stream, err := c.executionDataClient.SubscribeExecutionData(ctx, &req, opts...)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	sub := make(chan flow.ExecutionDataStreamResponse)
+	errChan := make(chan error)
+
+	go func() {
+		defer close(sub)
+		defer close(errChan)
+
+		for {
+			resp, err := stream.Recv()
+			if err == io.EOF {
+				return
+			}
+			if err != nil {
+				errChan <- fmt.Errorf("error receiving execution data: %w", err)
+				continue
+			}
+
+			execData, err := messageToBlockExecutionData(resp.GetBlockExecutionData())
+			if err != nil {
+				errChan <- fmt.Errorf("error converting execution data: %w", err)
+				return
+			}
+
+			//log.Printf("received execution data for block %d %x with %d chunks", resp.BlockHeight, execData.BlockID, len(execData.ChunkExecutionDatas))
+
+			sub <- flow.ExecutionDataStreamResponse{
+				Height:        resp.BlockHeight,
+				ExecutionData: execData,
+			}
+		}
+	}()
+
+	return sub, errChan, nil
+}
+
+func (c *BaseClient) SubscribeEvents(
+	ctx context.Context,
+	startBlockID flow.Identifier,
+	startHeight uint64,
+	filter flow.EventFilter,
+	opts ...grpc.CallOption,
+) (<-chan flow.BlockEvents, <-chan error, error) {
+
+	if startBlockID != flow.EmptyID && startHeight > 0 {
+		return nil, nil, fmt.Errorf("cannot specify both start block ID and start height")
+	}
+
+	req := executiondata.SubscribeEventsRequest{
+		Filter: &executiondata.EventFilter{
+			EventType: filter.EventTypes,
+			Address:   filter.Addresses,
+			Contract:  filter.Contracts,
+		},
+	}
+	if startBlockID != flow.EmptyID {
+		req.StartBlockId = startBlockID[:]
+	}
+	if startHeight > 0 {
+		req.StartBlockHeight = startHeight
+	}
+
+	stream, err := c.executionDataClient.SubscribeEvents(ctx, &req, opts...)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	sub := make(chan flow.BlockEvents)
+	errChan := make(chan error)
+
+	go func() {
+		defer close(sub)
+		defer close(errChan)
+
+		for {
+			resp, err := stream.Recv()
+			if err == io.EOF {
+				return
+			}
+			if err != nil {
+				errChan <- fmt.Errorf("error receiving execution data: %w", err)
+				continue
+			}
+
+			events := make([]flow.Event, len(resp.GetEvents()))
+			for _, ev := range resp.GetEvents() {
+				res, err := messageToEvent(ev, nil)
+				if err != nil {
+					errChan <- err
+					continue
```

Comment:
I don't think this error case should continue. If the message fails to decode, skipping here may result in missed events